src/app/components/component-viewer/component-usage/component-usage.component.ts

```diff
@@ -18,11 +19,37 @@ export class ComponentUsageComponent {
     }
 
     docLoaded() {
+        const preTags: Array<HTMLPreElement> = this.el.nativeElement.getElementsByTagName('pre');
+        for (const pre of preTags) {
+            pre.classList.add(pre.getElementsByTagName('code')[0].className.split('-')[1]);
+            this.removeLines(pre);
+            highlightBlock(pre);
+            this.addLines(pre);
+        }
+
         if ( this.section ) {
             const el = document.getElementById(this.section);
             if ( el ) {
                 el.scrollIntoView();
             }
         }
     }
+
+    private removeLines(pre: HTMLPreElement): void {
+        const span = pre.querySelector('span.line-number');
+        if (span) {
+            pre.removeChild(span);
+        }
+    }
+
+    private addLines(pre: HTMLPreElement): void {
```

Comment:
No way to reuse the new highlight directive to avoid repeating code here?

Reply:
yeah that's probably worth a look - I was getting pretty lazy by the end.  I'll see if there is a way to reuse.

Reply:
good suggestion @corykon - got that updated so the markdown directive and example viewer both leverage the HighlightDirective.  Was slightly tricky with the markdown parsing because the directive has to get added dynamically, but works just fine.  And now there aren't three duplicates of the same functions 👍